canopen/sdo/client.py

```diff
@@ -10,15 +11,140 @@
 
 from ..network import CanError
 
-from .base import SdoBase
 from .constants import *
 from .exceptions import *
+from ..objectdictionary import datatypes, Bits
+from ..objectdictionary.exceptions import OdIndexError, OdSubIndexError
 
 
 logger = logging.getLogger(__name__)
 
 
-class SdoClient(SdoBase):
+class DataProxy(object):
```

Comment:
We lose the ability to use Records and Arrays as dictionaries. It should for example be possible to iterate over them. Note that the length of an Array when using SDO must be read from subindex 0 in contrast to Arrays in the Object Dictionary. That's one of the reasons I had multiple implementations of those classes since they are similar but not the same.

It should also be possible to distinguish between variables, records and arrays in a straight forward way. This is an example of how it was done before:

```python
for item in node.sdo:
    if isinstance(item, canopen.sdo.Variable):
        print('%s = %d' % (item.od.name, item.raw))
    elif isinstance(item, canopen.sdo.Array):
        print('%s (array):' % item.od.name)
        for array_item in item:
            print('    %d: %d' % (array_item.od.subindex, array_item.raw))
    elif isinstance(item, canopen.sdo.Record):
        print('%s (record):' % item.od.name)
        for record_item in item:
            print('    %s: %d' % (record_item.od.name, record_item.raw))

Reply:
Ok I have no quick answer to this. I need to think a bit...

Reply:
I could accept the pull request as it is, but I might do some changes of my own afterwards.